### PR TITLE
Add docs for reference commands

### PR DESCRIPTION
**Pull request by kwmonroe**
### Summary

Generate reference docs for `k8s $cmd` and add them to our `./docs/src/_parts`.

### Changes

- un-hide `k8s config` and `k8s status` commands
- new `gen_cmd_docs.go` that uses cobra to generate markdown based on our `./src/k8s/cmd/k8s` commands
- new `go.doc` make target that runs as part of the code quality gh workflow
  - if cmd docs are affected but not included in future PRs, this action will fail telling authors to run `make go.doc`
- include initial `./docs/src/_parts` content

### CODE REVIEW

**Comment by evilnick** [comment]
Cool. I have a trick for these docs, let me see if I can make it work

**Comment by evilnick** [comment]
cool, that works! Not crazy about some of the formatting, but I can probably strip some of it out.
Do these get regenerated with the build?

File: src/k8s/cmd/k8s/k8s.go [hunk 1]
```diff
@@ -71,5 +71,6 @@ func NewRootCmd() *cobra.Command {
 	rootCmd.AddCommand(newHelmCmd())
 	rootCmd.AddCommand(newKubectlCmd())
 
+	rootCmd.DisableAutoGenTag = true
```
Comment (kwmonroe) [inline src/k8s/cmd/k8s/k8s.go:74]: without this, we get this at the bottom of our docs:

`generated by cobra`

which isn't bad, per se.. just not necessary in our reference docs.

File: src/k8s/Makefile [hunk 2]
```diff
@@ -6,6 +6,10 @@ go.fmt:
 	go mod tidy
 	go fmt ./...
 
+go.docs: go.doc
+go.doc:
+	$(DQLITE_BUILD_SCRIPTS_DIR)/static-go-doc.sh gen_cmd_docs.go
+
```
Comment (kwmonroe) [inline src/k8s/Makefile:12]: `go run gen_cmd_docs.go` needs `dqlite.h`

File: src/k8s/hack/gen_cmd_docs.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package main
```
Comment (kwmonroe) [inline src/k8s/hack/gen_cmd_docs.go:1]: this impl of cobra needs to be `go run`, which needs a `main`, which would conflict with our `./src/k8s/cmd/main.go` -- hence this is in our `hack` util dir.

**Comment by evilnick** [comment]
This gets a :cake: from me for how the docs turn out - not sure if anyone wants to review the code changes

File: src/k8s/hack/gen_cmd_docs.go [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+)
+
+func main() {
+	k8sCmd := k8s.NewRootCmd()
+
+	err := doc.GenMarkdownTree(k8sCmd, "../../../docs/src/_parts/commands")
+	if err != nil {
+		log.Fatal(err)
+	}
+}
```
Comment (neoaggelos) [inline src/k8s/hack/gen_cmd_docs.go:17]: could we make this a hidden `k8s documentation` command? That way you can avoid the `-C ${DIR}` things below

Comment (kwmonroe) [reply src/k8s/hack/gen_cmd_docs.go:17]: i can make this a `k8s $cmd` and generate docs at runtime, but i'm not sure how to call it from a Makefile / gh action that can determine if a PR-under-test is missing docs.

Comment (kwmonroe) [reply src/k8s/hack/gen_cmd_docs.go]: aha!  call it like you would any other command, so long as it's built/linked in the go.doc target.  thanks @neoaggelos!

**Review by kwmonroe — COMMENTED** [review]
LGTM with some clutch plays from the team.

**Pull request merged**